src/util/iterative_restoration.jl

```diff
@@ -171,31 +234,52 @@ function merge_solution!(solution1, solution2)
     end
 end
 
-
-" Update damage status for each time period based on whether the device has already been repaired"
-function update_damage_status!(mn_data)
+function _get_item_repairs(mn_data)
+    repairs = Dict{String,Array{Tuple{String,String},1}}(nw=>[] for nw in keys(mn_data["nw"]))
     if _IM.ismultinetwork(mn_data)
         for (nw_id, network) in mn_data["nw"]
-            for (comp_type, comp_status) in _PM.pm_component_status
-                for (comp_id, comp) in network[comp_type]
+            for comp_name in restoration_comps
+                status_key = _PM.pm_component_status[comp_name]
+                for (comp_id, comp) in network[comp_name]
                     if nw_id != "0" #not items are repaired in "0", do not check in previous network for a change
-                        if comp[comp_status] != _PM.pm_component_status_inactive[comp_type] &&  # if comp is active
-                            mn_data["nw"]["$(parse(Int,nw_id)-1)"][comp_type][comp_id][comp_status] != _PM.pm_component_status_inactive[comp_type] # if comp was previously active
-                            if haskey(comp,"damaged") && comp["damaged"] == 1
-                                # therefore the comp was repaired in a former time_step, should be considered undamaged
-                                Memento.info(_PM._LOGGER, "$(comp_type) $(comp_id) was repaired before step $(nw_id). Setting damged state to 0.")
-                                comp["damaged"] = 0
-                            end
+                        if comp[status_key] != _PM.pm_component_status_inactive[comp_name] &&  # if comp is active
+                            mn_data["nw"]["$(parse(Int,nw_id)-1)"][comp_name][comp_id][status_key] == _PM.pm_component_status_inactive[comp_name] # if comp was previously inactive
+                            push!(repairs[nw_id], (comp_name,comp_id))
                         end
                     end
                 end
             end
         end
     else
-        Memento.error(_PM._LOGGER, "update_damage_status required multinetwork to identify is a device has been previously repaired.")
+        Memento.error(_PM._LOGGER, "get_item_repairs requires multinetwork.")
```

Comment:
Purely a stylistic thing, when your function checks conditions and errors if they are not meet, I would put them at the top of the function like so,
```
function
if !check_1
   error_1
end
if !check_2
   error_2
end

core code...

end
```

Instead of,
```
function
if !check_1
   if !check_2
     core code...
   else
      error2
   end
else
   error_1
end
return
end
```